includes/abstracts/abstract-wc-order.php

```diff
@@ -77,235 +80,157 @@
+	 *
 	 * @since 2.7.0
 	 * @var array
 	 */
 	protected $items_to_delete = array();
 
 	/**
-	 *  Internal meta type used to store order data.
+	 * Internal meta type used to store order data.
+	 *
 	 * @var string
 	 */
 	protected $meta_type = 'post';
 
 	/**
 	 * Stores meta in cache for future reads.
+	 *
 	 * A group must be set to to enable caching.
 	 * @var string
 	 */
 	protected $cache_group = 'order';
 
 	/**
+	 * Which data store to load.
+	 *
+	 * @var string
+	 */
+	protected $data_store_name = 'order';
+
+	/**
 	 * Get the order if ID is passed, otherwise the order is new and empty.
 	 * This class should NOT be instantiated, but the get_order function or new WC_Order_Factory.
 	 * should be used. It is possible, but the aforementioned are preferred and are the only.
 	 * methods that will be maintained going forward.
 	 *
-	 * @param  int|object|WC_Order $read Order to init.
+	 * @param  int|object|WC_Order $order Order to read.
 	 */
-	public function __construct( $read = 0 ) {
-		parent::__construct( $read );
+	public function __construct( $order = 0 ) {
+		parent::__construct( $order );
 
-		if ( is_numeric( $read ) && $read > 0 ) {
-			$this->read( $read );
-		} elseif ( $read instanceof self ) {
-			$this->read( absint( $read->get_id() ) );
-		} elseif ( ! empty( $read->ID ) ) {
-			$this->read( absint( $read->ID ) );
+		if ( is_numeric( $order ) && $order > 0 ) {
+			$this->set_id( $order );
+		} elseif ( $order instanceof self ) {
+			$this->set_id( absint( $order->get_id() ) );
+		} elseif ( ! empty( $order->ID ) ) {
+			$this->set_id( absint( $order->ID ) );
+		} else {
+			$this->set_object_read( true );
 		}
+
 		// Set default status if none were read.
 		if ( ! $this->get_status() ) {
 			$this->set_status( apply_filters( 'woocommerce_default_order_status', 'pending' ) );
 		}
-	}
 
-	/*
-	|--------------------------------------------------------------------------
-	| CRUD methods
-	|--------------------------------------------------------------------------
-	|
-	| Methods which create, read, update and delete orders from the database.
-	| Written in abstract fashion so that the way orders are stored can be
-	| changed more easily in the future.
-	|
-	| A save method is included for convenience (chooses update or create based
-	| on if the order exists yet).
-	|
-	*/
+		$this->data_store = WC_Data_Store::load( $this->data_store_name );
+
+		if ( $this->get_id() > 0 ) {
+			$this->data_store->read( $this );
+		}
+	}
 
 	/**
-	 * Get internal type (post type.)
+	 * Get internal type.
+	 *
 	 * @return string
 	 */
 	public function get_type() {
 		return 'shop_order';
 	}
 
 	/**
+	 * Prefix for action and filter hooks on data.
+	 *
+	 * @since  2.7.0
+	 * @return string
+	 */
+	protected function get_hook_prefix() {
+		return 'woocommerce_get_order_';
+	}
+
+	/**
 	 * Get a title for the new post type.
+	 *
+	 * @return string
 	 */
-	protected function get_post_title() {
+	public function get_post_title() {
```

Comment:
And maybe change the name of this function too, since when change orders to a database, this will not make more sense.